qp_klp/klp.py

```diff
@@ -282,13 +272,15 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
         if skip_exec:
             cmds = []
 
+        cmd_log_fp = join(out_dir, 'cmds.log')
+        with open(cmd_log_fp, 'w') as cmd_log_f:
+            for cmd in cmds:
+                cmd_log_f.write(f'{cmd}\n')
+
         for cmd in cmds:
             p = Popen(cmd, universal_newlines=True, shell=True,
                       stdout=PIPE, stderr=PIPE)
             std_out, std_err = p.communicate()
-            return_code = p.returncode
-            if return_code != 0:
```

Comment:
Just to confirm, why isn't this required anymore? AFAIK this is good to have ...

Reply:
I would say in general it's good, but in practice it depends on how well-behaved the command being called is. This goes back to the recent issue where tar returned 1 ('cd /redacted/redacted/working_dir/d6624327-bd4e-4702-9bb2-06597dce7892; tar zcvf logs-FastQCJob.tgz FastQCJob/logs') even though logs-FastQCJob.tgz was properly generated and the unarchived files were indistinguishable from a manually generated tgz file that properly returned 0.

For an entire run to error out after successfully iterating through all of the stages seems misaligned. At best the system should return a warning, rather than an error.

Reply:
I'll say, let's readd it ...

Reply:
and return a warning?

Reply:
Nah, I'll say break the full job cause this was actually a bug in the storage, which was fixed this morning.

Reply:
okay.